library_generation/utils/utilities.sh

```diff
@@ -157,7 +157,22 @@ download_generator_artifact() {
 download_protoc() {
   local protoc_version=$1
   local os_architecture=$2
-  if [ ! -d "protoc-${protoc_version}" ]; then
+
+  protoc_dirname="protoc-${protoc_version}"
+  protoc_path="${output_folder}/${protoc_dirname}/bin"
+  if [[ -f "${protoc_path}/protoc" ]]; then
+    return
+  fi
+
+  if [[ "${DOCKER_PROTOC_VERSION}" == "${protoc_version}" ]]; then
```

Comment:
The protoc version priority:
1. version in generation config.
2. version in image.
3. inferred from the generator.

Reply:
Added this logic to `get_protoc_version`